src/components/WalletConnect/NanoContract/NanoContractMethodArgs.js

```diff
@@ -129,6 +146,65 @@ export const NanoContractMethodArgs = ({ blueprintId, method, ncArgs }) => {
   )
 };
 
+/**
+ * Component resposible to render the appropriate format for the value
+ * taking in consideration the type.
+ *
+ * Remarks
+ * The values received in here when derived from 'byte' type like
+ * 'TxOutputScript', 'TokenUid' and 'VertexId' are already in its
+ * hexadecimal format.
+ *
+ * Values of type 'Address', which also derives from 'byte' are
+ * in its base58 format.
+ *
+ * Values of type 'SignedData[Result]' arrives here using its data
+ * part already.
+ *
+ * @param {Object} props
+ * @param {string} props.type An argument type
+ * @param {string} props.value An argument value
+ * @param {Object} props.network A network object
+ * @param {Object} props.tokens A map of registered tokens
+ * @param {Object} props.deserializer The Nano Contract deserializer
+ */
+const ArgValue = ({ type, value, network, tokens, deserializer }) => {
+  if (type === 'Amount') {
+    return renderValue(value);
+  }
+
+  if (type === 'Timestamp') {
+    return getTimestampFormat(value);
+  }
+
+  if (type === 'TxOutputScript') {
+    const parsedScript = parseScriptData(value, network);
+    if (parsedScript && parsedScript.getType() === 'data') {
+      return parsedScript.data;
+    }
+
+    if (parsedScript) {
+      return parsedScript.address.base58;
+    }
+  }
+
+  if (type === 'TokenUid') {
+    if (value === DEFAULT_TOKEN.uid) {
+      return DEFAULT_TOKEN.symbol;
+    }
+
+    if (value in tokens) {
+      return tokens[value].symbol;
+    }
+  }
+
+  if (type === 'Address') {
+    return deserializer.toAddress(Buffer.from(value, 'hex'));
+  }
```

Comment:
Do you need to do this with `Address` type? I thought it would come as base58 from the RPC lib.

Reply:
Yes, I need it. It is coming as a string in hexadecimal from RPC.

Reply:
It's actually coming from the dApp as hex, this will be fixed in the new dApp

Reply:
I'm wondering if this is a dApp responsibility. Is it?

Reply:
I removed the Address parse because the dApp is already handling it.